app/models/probate.rb

```diff
@@ -1,5 +1,29 @@
 class Probate < Base
   attribute :kase, Boolean
+  attribute :deceased_name, String
+  attribute :date_of_death, Date
+
+  TIME_LIMIT_FOR_PROBATE = 20
 
   validates :kase, inclusion: { in: [true, false] }
+
+  with_options if: :kase do
+    validates :deceased_name, presence: true
+  end
+
+  with_options if: :validate_probate_date_of_death? do
+    validates :date_of_death, date: {
+      after_or_equal_to: TIME_LIMIT_FOR_PROBATE.years.ago,
+      before: Time.zone.tomorrow,
+      allow_blank: false
+    }
+  end
+
+  private
+
+  def validate_probate_date_of_death?
+    if kase?
+      validate_date? 'date_of_death'
+    end
```

Comment:
But really, then you could just be doing:

```ruby
with_options if: :kase? do
```
instead of

```ruby
with_options if: :validate_probate_date_of_death? do
```